modules/android/pages/vector-search-api-reference.adoc

```diff
@@ -0,0 +1,286 @@
+
+|===
+
+This function performs vector search against a specific vector index identifier for the specified vector expression.
+If the specified index does not exist, an error will occur on creation of the query.
+The matched vectors will be returned up to the specified limit number, if the limit is not specified then the default value will be used. 
+The returned vectors are sorted by their distance values in ascending order by default.
+
+NOTE: The default value for the `limit` parameter is 3.
+
+[source, sqlpp]
+----
+SELECT name 
+FROM photos 
+WHERE vector_match(photos-index, [0.1, 0.5, 1.2])
+
+----
+
+[source, sqlpp]
+----
+
+SELECT name 
+FROM photos
+WHERE vector_match(photos-index, [0.1, 0.5, 1.2]) AND city = "San Francisco"
+
+----
+
+=== Vector Distance
+
+==== `vector_distance(vectorIndexIdentifier)`
+
+[cols = "3*"]
+|===
+|Parameter |Is Required |Description
+
+|vectorIndexIdentifier
+|Required
+|The name of the vector index.
+
+|===
+
+This function returns the distance between the target vector specified in the `vector_match()` function and the matched vector in the specified vector index based on the distance metric set in the index configuration .
+
+[source, sqlpp]
+----
+
+SELECT name, vector_distance(photos-index) 
+FROM photos
+WHERE vector_match(photos-index, [0.1, 0.5, 1.2])
+
+----
+
+[source, sqlpp]
+----
+
+SELECT name WHERE vector_match(photos-index, [0.1, 0.5, 1.2]) 
+FROM photos
+ORDER BY vector_distance(photos-index) DSC
+
+----
+
+IMPORTANT: Similar to the Full Text Search `match()` function, `vector_match()` can only be called alone or at the top level `AND` expression.
+
+=== Generate Vectors with Prediction
+
+You can use two methods to generate vectors in Couchbase Lite:
+
+. You can call a Machine Learning(ML) model, and embed the generated vectors inside the documents.
+
+. You can use the `prediction()` function to generate vectors to be indexed for each document at the indexing time.
+
+Below is an example configuration.
+
+[source, c]
+----
+
+let config = VectorIndexConfiguration(expression: "prediction(photo-model, photo).vector", dimensions: 256, centroids: 30)
+try collection.createIndex(withName: "photo-index", config: config)
+
+----
+
+You can use less storage by using the `prediction()` function as the encoded vectors will only be stored in the index. 
+However, the index time will be longer as indexing is occurring at run time.
+
+[IMPORTANT]
+--
+
+In the beta release. Updating the index for the documents whose returned vectors from the `prediction()` were missing on calling the ML model is not possible unless you perform the following actions:
+
+* Update the documents
+
+* Regenerate the index
+
+--
+
+== See Also
+
+* xref:swift:vector-search.adoc[Vector Search]
+
+* xref:swift:working-with-vector-search.adoc[Working with Vector Search]
+
+* xref:swift:fts.adoc[Full Text Search]
+
```

Comment:
General- we must not duplicate what is in API reference guides here (expectation is that we will publish those when we go Beta). We must only include code snippets describing usage and then link to API reference guide for correct API definitions.